eng/pipelines/templates/jobs/smoke-tests.yml

```diff
@@ -107,10 +110,38 @@ jobs:
           pwsh: true
           ${{ if eq(parameters.Daily, true) }}:
             arguments: -CI -Daily
-            displayName: Use latest dev feed package versions
+            displayName: Use latest daily package versions
           ${{ if eq(parameters.Daily, false) }}:
             arguments: -CI
-            displayName: Use latest package versions
+            displayName: Use latest GA/Beta package versions
+
+      - ${{ if eq(parameters.Daily, false) }}:
+        - download: current
+          artifact: ${{ parameters.ArtifactName }}
+          timeoutInMinutes: 5
+
+        - ${{ each artifact in parameters.Artifacts }}:
+          - ${{ if and(ne(variables['Skip.Release'], 'true'), ne(artifact.skipPublishPackage, 'true')) }}:
+            - pwsh: |
+                [xml]$csproj = Get-Content ./common/SmokeTests/SmokeTest/SmokeTest.csproj
+                $deps = ($csproj | Select-Xml -Xpath //Project/ItemGroup/PackageReference).Node.Include
+
+                if ($deps.Contains("${{ artifact.name }}")) {
+                  $files = Get-ChildItem "$(Pipeline.Workspace)/${{ parameters.ArtifactName }}/${{ artifact.name }}/*.nupkg"
```

Comment:
May not be needed here, though, because you'd not otherwise get into this conditional block if the one above failed.